File: observer/c-sharp/Samsung.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Observer
{
    // Observer sınıfından türer.
    // UML diyagramındaki ConcreteObservable yapısına denk gelmektedir.
    class Samsung : ObservableProduct
    {
        // Ürün fiyatı değiştikten sonra Observer örneklerine bildirim gönderilir.
        public void ChangePrice()
        {
            // Ürün fiyatlarının güncellenmesi vs.

            this.message.productName = "Samsung";
            this.message.message = "Samsung's price updated.";
            this.NotifyObserver();
        }
    }
}

File: observer/go/samsung.go
package main

// Observer sınıfından türer.
// UML diyagramındaki ConcreteObservable yapısına denk gelmektedir.
type samsung struct {
	ObservableProduct *ObservableProduct
}

func NewSamsung() *samsung {
	return &samsung{
		ObservableProduct: NewObservableProduct(),
	}
}

// ChangePrice Ürün fiyatı değiştikten sonra Observer örneklerine bildirim gönderilir.
func (s *samsung) ChangePrice() {
	// Ürün fiyatlarının güncellenmesi vs.
	s.ObservableProduct.message.ProductName = "Samsung"
	s.ObservableProduct.message.Message = "Samsung's price updated."
	s.ObservableProduct.NotifyObserver()
}
